eng/source-build-patches/0029-libraries-packages-use-net5.0-PackagingTaskA.patch

```diff
@@ -0,0 +1,25 @@
+From da5a2c2593df373d1fafa605e1d7ce3ba2ca38e7 Mon Sep 17 00:00:00 2001
+From: Tom Deseyn <tom.deseyn@gmail.com>
+Date: Wed, 18 Nov 2020 09:29:17 +0100
+Subject: [PATCH] runtime: libraries-packages: use net5.0 PackagingTaskAssembly
+
+---
+ src/libraries/libraries-packages.proj | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/libraries/libraries-packages.proj b/src/libraries/libraries-packages.proj
+index 603d523df55..38cd7e74337 100644
+--- a/src/libraries/libraries-packages.proj
++++ b/src/libraries/libraries-packages.proj
+@@ -7,7 +7,7 @@
+   
+   <PropertyGroup>
+     <PackagingTaskAssembly>$(NuGetPackageRoot)microsoft.dotnet.build.tasks.packaging\$(MicrosoftDotNetBuildTasksPackagingVersion)\tools\</PackagingTaskAssembly>
+-    <PackagingTaskAssembly Condition="'$(MSBuildRuntimeType)' == 'core'">$(PackagingTaskAssembly)netcoreapp3.1\</PackagingTaskAssembly>
++    <PackagingTaskAssembly Condition="'$(MSBuildRuntimeType)' == 'core'">$(PackagingTaskAssembly)net5.0\</PackagingTaskAssembly>
+     <PackagingTaskAssembly Condition="'$(MSBuildRuntimeType)' != 'core'">$(PackagingTaskAssembly)net472\</PackagingTaskAssembly>
+     <PackagingTaskAssembly>$(PackagingTaskAssembly)Microsoft.DotNet.Build.Tasks.Packaging.dll</PackagingTaskAssembly>
```

Comment:
This usage can be deleted if this project just uses the package's props to define the task: https://github.com/dotnet/arcade/blob/13bb7c843fac9ce3652b5f418ca27b9bf3b83754/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.common.targets#L44

Reply:
This is currently necessary because of https://github.com/dotnet/arcade/issues/7413

Reply:
I see, should be simple enough to write those targets with a template task?   At the very least, why isn't this patch in Arcade instead?

Reply:
> At the very least, why isn't this patch in Arcade instead?

What would you patch in Arcade? For whatever reason, the `Microsoft.DotNet.Build.Tasks.Packaging` package doesn't expose a property that points to the msbuild task assembly.

I wouldn't want to invest any resources in fixing this aside from inlining the patch into main as this code-path will go away  when @Anipik finished the pkgproj migration.

Reply:
> This usage can be deleted if this project just uses the package's props to define the task

The project can't use the package's props or targets files as it contributes to the project's restore. This runs as part of restore.

Reply:
> What would you patch in Arcade

My impression based on your comment `This is currently necessary because of dotnet/arcade#7413` was that the props/targets didn't have the right path.

If the props have the right path, then this proj file can just import the props/targets.  Since all this proj does is [define a task](https://github.com/dotnet/runtime/blob/9199eb6952a4569cf5be57b68732989651ca55e2/src/libraries/libraries-packages.proj#L34) which those targets [already define.](https://github.com/dotnet/arcade/blob/36b148348ee8312f6369c0c56b0d0fe07deec603/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.common.targets#L44)

Reply:
> The project can't use the package's props or targets files as it contributes to the project's restore.

It should be downloaded before that, I think, as part of https://github.com/dotnet/runtime/blob/9199eb6952a4569cf5be57b68732989651ca55e2/eng/Tools.props#L9.  

I'd prefer to make things less hacky if possible though, was hoping your conversion of this to a traversal project would just let it naturally consume the package and let us delete stuff.  If that's not the case then I'm fine leaving this since it'll go away with PkgProj removal, as you said.

Reply:
> It should be downloaded before that, I think, as part of

Right, that's how the package is available before the project's restore to calculate the PackageDownload items. Libraries-packages.proj itself would need to manually import the props and targets files manually which IMO isn't ideal either. Let's just keep it as-is for now and apply the patch.